src/app/projects/components/project-settings/components/labeling-tasks/labeling-tasks.component.ts

```diff
@@ -0,0 +1,304 @@
+        }
+      });
+  }
+
+  isLabelingTaskOptionDisabled(task: AbstractControl, dropdownValue: string) {
+    const targetID = task.get('targetId').value;
+    if (
+      targetID != '' &&
+      dropdownValue == LabelingTask.INFORMATION_EXTRACTION
+    ) {
+      if (this.attributeAlreadyHasInformationExtraction(targetID)) return true;
+      else if (this.dataHandlerHelper.getAttributeArrayAttribute(targetID, 'dataType', this.attributes) != 'TEXT')
+        return true;
+    } else if (
+      targetID == '' &&
+      dropdownValue == LabelingTask.INFORMATION_EXTRACTION
+    )
+      return true;
+    return false;
+  }
+
+  isTaskNameUniqueCheck(name: string, ownGroup: FormGroup = null): boolean {
+    if (name == '') return true;
+    const nameToExclude = ownGroup ? ownGroup.get("name").value : "";
+    let taskName;
+    for (let task of this.labelingTasksArray.controls) {
+      taskName = task.get('name').value;
+      if (name == taskName && taskName != nameToExclude) return false;
+    }
+    return true;
+  }
+
+  isLabelNameUnique(taskId: string, name: string): boolean {
+    return this.lh.isLabelNameUnique(taskId, name);
+  }
+
+  setLabelingTaskTarget(id: string) {
+    this.settingModals.labelingTask.create.taskId = id;
+  }
+
+  checkLabelingTaskName(eventTarget: HTMLInputElement) {
+    this.isTaskNameUnique = this.isTaskNameUniqueCheck(eventTarget.value);
+    this.settingModals.labelingTask.create.name = eventTarget.value;
+  }
+
+  attributeAlreadyHasInformationExtraction(attributeId: string): boolean {
+    for (let task of this.labelingTasksArray.controls) {
+      if (attributeId == task.get('targetId').value) {
+        if (task.get('taskType').value == LabelingTask.INFORMATION_EXTRACTION)
+          return true;
+      }
+    }
+    return false;
+  }
+
+  checkAndModifyLabelName(eventTarget: HTMLInputElement) {
+    eventTarget.value = eventTarget.value.replace("-", " ");
+    this.settingModals.label.create.labelName = eventTarget;
+  }
+
+  onlyLabelsChanged(tasks: any): boolean {
+    if (this.labelingTasksArray.controls.length == 0) return false;
+    if (this.labelingTasksArray.controls.length != tasks.length) return false;
+    for (const task of tasks) {
+      if (this.getTaskArrayAttribute(task.id, 'id') == 'UNKNOWN') return false;
+      if (this.getTaskArrayAttribute(task.id, 'taskType') != task.taskType)
+        return false;
+      if (this.getTaskArrayAttribute(task.id, 'name') != task.name)
+        return false;
+    }
+
+    return true;
+  }
+
+  getTaskArrayAttribute(taskId: string, valueID: string) {
+    for (let task of this.labelingTasksArray.controls) {
+      if (taskId == task.get('id').value) return task.get(valueID).value;
+    }
+    return 'UNKNOWN';
+  }
+
+  removeLabel() {
+    const labelDeleteData = this.settingModals.label.delete;
+    this.lh.removeLabel(this.project.id, labelDeleteData.taskId, labelDeleteData.label.id, labelDeleteData.label.color);
```

Comment:
why does a remove function need a label color? also I think we should at least check for modal open that if it's not open we don't remove a random label

Reply:
found the reason for oldcolor (so the new labels know what colors are already taken)